src/localization/online_localizer/online_localizer.cpp

```diff
@@ -263,21 +225,24 @@ bool OnlineLocalizer::nodeWorthExpanding(const Node &node) const {
 
 void OnlineLocalizer::updateSearch(const NodeSet &successors) {
   Node possibleHyp = getProminentSuccessor(successors);
-  // printf("[DEBUG][OnlineLocalizer] Prominent child is: ");
-  // possibleHyp.print();
 
-  if (possibleHyp.quId > _currentBestHyp.quId) {
-    _currentBestHyp = possibleHyp;
-  } else if (possibleHyp.quId == _currentBestHyp.quId) {
+  if (possibleHyp.quId > currentBestHyp_.quId) {
+    // The search is reaching a query further in sequence than current best
+    // hypothesis. This is always better.
+    currentBestHyp_ = possibleHyp;
+  } else if (possibleHyp.quId == currentBestHyp_.quId) {
+    // The alternative path to the same level as current best hypothesis as
+    // found.
     double accCost_current =
-        _accCosts[_currentBestHyp.quId][_currentBestHyp.refId];
-    double accCost_poss = _accCosts[possibleHyp.quId][possibleHyp.refId];
+        accCosts_[currentBestHyp_.quId][currentBestHyp_.refId];
+    double accCost_poss = accCosts_[possibleHyp.quId][possibleHyp.refId];
     if (accCost_poss <= accCost_current) {
-      _currentBestHyp = possibleHyp;
+      // Accumulated cost of new possible hypothesis is smaller than accumulated
+      // cost for the current best hypothesis. This becomes new current best
+      // hypothesis.
```

Comment:
If you expand the names of your variables the code will say _exactly_ the same thing as your comment does here. I think having longer variables is better for readability and then there will be less need for lengthy comments that tend to go out of sync with the code with time.